### PR TITLE
Check in canary, useful for debugging

### PR DESCRIPTION
**Pull request by tylerwilliams**

### CODE REVIEW

File: server/util/canary/canary.go [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// StartWithLateFn starts a canary that will call lateFn every expectedDuration
+// period (if not cancelled) and once when cancelled.
+func StartWithLateFn(expectedDuration time.Duration, lateFn CanaryFunc, doneFn CanaryFunc) CancelFunc {
+	if !*enableCanaries {
+		return func() {}
+	}
+
+	done := make(chan struct{})
+	cancel := func() {
+		close(done)
```
Comment (vadimberezniker) [inline server/util/canary/canary.go:28]: can you guard against cancel being called multiple times?

File: server/util/canary/canary.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
+package canary
```
Comment (vadimberezniker) [inline server/util/canary/canary.go:1]: watchdog is another name to consider if you're looking for alternatives

**Review by vadimberezniker — APPROVED** [review]

File: server/util/canary/canary.go (hunk 1 above)
Comment (tylerwilliams) [reply server/util/canary/canary.go]: Done.

File: server/util/canary/canary.go (hunk 2 above)
Comment (tylerwilliams) [reply server/util/canary/canary.go:1]: Yeah I considered that, but I always think of a watchdog timer where you have to call reset on it or it'll restart your thing. This is sort of the opposite of that in a way?

**Pull request merged**